data_provenance/to-dolma.py

```diff
@@ -0,0 +1,148 @@
+"""Convert the index+data into the dolma sharded jsonl.gz format."""
+
+import argparse
+import functools
+import gzip
+import json
+import operator as op
+import os
+from datetime import datetime
+
+import jsonlines
+import pandas as pd
+
+from data_provenance.constants import HF_MAPPING
+from licensed_pile.licenses import PermissiveLicenses
+
+# from licensed_pile.write import to_dolma
+
+LICENSE_MAPPER = {
+    "CDLA Sharing 1.0": PermissiveLicenses.CDLA,
+    "MPL 2.0": PermissiveLicenses.MPL,
+    "CDLA Permissive 1.0": PermissiveLicenses.CDLA_P,
+    "Custom": PermissiveLicenses.CUSTOM,
+    "No License": PermissiveLicenses.NO_LICENSE,
+    "OANC": PermissiveLicenses.OANC,
+    "GNU General Public License v3.0": PermissiveLicenses.GPL_V3,
+    "MIT License": PermissiveLicenses.MIT,
+    "CC BY 4.0": PermissiveLicenses.CC_BY,
+    "CC0 1.0": PermissiveLicenses.CC0,
+    "BSD 2-Clause License": PermissiveLicenses.BSD,  # Assuming "BSD" refers to both 2-Clause and 3-Clause
```

Comment:
These are already being merged in the existing implementation. I'm not sure what to set the original `PermissiveLicenses.BSD` to 2 or 3?

Reply:
AFAIK, no of the current datasets currently use BSD and I was just enumerating licenses I knew off the top of my head originally lol. So I think you should be fine to split into 2 and 3, can probably also remove the `.BSD` in favor of `.BSD_2` and `.BDS_3` as I don't think it will break things.